src/utils/prim-parsing.fs

```diff
@@ -234,8 +231,14 @@ module internal Implementation =
         let ruleValues    = (Array.zeroCreate 100 : obj[])              
         let lhsPos        = (Array.zeroCreate 2 : Position[])                                            
         let reductions = tables.reductions
-        let actionTable = new AssocTable(tables.actionTableElements, tables.actionTableRowOffsets)
-        let gotoTable = new AssocTable(tables.gotos, tables.sparseGotoTableRowOffsets)
+        let cacheSize = 7919 // the 1000'th prime
+        // Use a simpler hash table with faster lookup, but only one
+        // hash bucket per key.
+        let actionTableCache = ArrayPool<int>.Shared.Rent(cacheSize * 2)
+        let gotoTableCache = ArrayPool<int>.Shared.Rent(cacheSize * 2)
+        use _cacheDisposal = { new IDisposable with member _.Dispose() = ArrayPool<int>.Shared.Return actionTableCache; ArrayPool<int>.Shared.Return gotoTableCache }
```

Comment:
Style - let's not do this on one line, also multi-line should make this more debuggable

Reply:
Sure